modules/videoio/src/cap_v4l.cpp

```diff
@@ -589,8 +594,14 @@ static void v4l2_create_frame(CvCaptureCAM_V4L *capture) {
     }
 
     /* Set up Image data */
-    cvInitImageHeader(&capture->frame, size, IPL_DEPTH_8U, channels);
+    switch(capture->palette) {
+    case V4L2_PIX_FMT_Y16:
+        cvInitImageHeader(&capture->frame, size, IPL_DEPTH_16U, channels);
```

Comment:
Yeah, that makes sense. No need to set the size, it seems, channels should suffice.